### PR TITLE
Storybook docs structure

### PR DESCRIPTION
**Pull request by danielleflinn**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [X ] Other-
```

## Objective
Update the Storybook grouping structure to to allow for better organization of component documentation.
- updated 'commons' grouping to 'documentation'
-  added a mdx file for button docs 
- updated sort order to include docs before component stories

## Code changes

- **introduction.stories.mdx.** updated meta name and typo
- **colors.stories.mdx.** updated meta name 
- **icons.stories.mdx.** updated meta name 
- **buttons.stories.mdx.** added to display component in the new documentation section
- **.storybook/preview.js** updated [story sort order](https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy#sorting-stories) to keep docs at the top and introduction as the landing page. 

## Screenshots
Screen Shot 2022-10-11 at 5 16 12 PM

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: libs/components/src/stories/colors.stories.mdx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Meta } from "@storybook/addon-docs";
 
-<Meta title="Common/Colors" />
+<Meta title="Documentation/Colors" />
```
Comment (danielleflinn) [inline libs/components/src/stories/colors.stories.mdx:3]: @bitwarden/dept-design I've created this draft PR so we can nail down the title for the documentation pages. We discussed using "Documentation" last week so I've added that here as a placeholder. Does this work for you all?

File: .storybook/preview.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ export const parameters = {
       date: /Date$/,
     },
   },
+  options: {
+    storySort: {
+      order: ["Documentation", ["Introduction", "Colors", "Icons"], "Component Library"],
+    },
+  },
```
Comment (coroiu) [inline .storybook/preview.js:19]: Oh I didn't know you could do this! Nice!

File: libs/components/src/stories/buttondocs.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Accessibility
+
+### Color contrast
+
+All button styles are WCAG compliant when displayed on `background` and `background-alt` colors. To use a button on a different background, double check that the color contrast is sufficient in both the light and dark themes.
+
+### Loading Buttons
+
+Include an `aria-label` attribute that defaults to “loading” but can be configurable per implementation. On click, the screen reader should announce the `aria-label`. Once the action is compelted, use another messaging pattern to alert the user that the action is complete (example: success toast).
```
Comment (coroiu) [inline libs/components/src/stories/buttondocs.stories.mdx:73]: thought(non-blocking): I don't think we're changing the `label` when the buttons transitions to the loading state. We probably have to update the button components with this logic

**Review by coroiu — APPROVED** [review]
Very excited about this change!

File: .storybook/preview.js [hunk 4]
```diff
@@ -12,6 +12,11 @@ export const parameters = {
       date: /Date$/,
     },
   },
+  options: {
+    storySort: {
+      order: ["Documentation", ["Introduction", "Colors", "Icons"], "Component Library"],
```
Comment (Hinton) [inline .storybook/preview.js:17]: What happens if we add more sub-pages to documentation?

File: libs/components/src/stories/buttondocs.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
+import { Meta, Story, Source } from "@storybook/addon-docs";
```
Comment (Hinton) [inline libs/components/src/stories/buttondocs.stories.mdx:1]: Can we add a dash in the name, `button-docs.stories.mdx`

File: .storybook/preview.js (hunk 4 above)
Comment (danielleflinn) [reply .storybook/preview.js:17]: Any pages that are added not called out in the subpage array there will come after the 'icons' page in alphabetical order.

File: libs/components/src/stories/buttondocs.stories.mdx (hunk 5 above)
Comment (danielleflinn) [reply libs/components/src/stories/buttondocs.stories.mdx:1]: For readability? Sure. I'll add a not to our documentation that all doc stories should be named this way.

File: libs/components/src/stories/Introduction.stories.mdx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { Meta } from "@storybook/addon-docs";
 
-<Meta title="Common/Introduction" />
+<Meta title="Documentation/Introduction" />
```
Comment (jtouchstonebw) [inline libs/components/src/stories/Introduction.stories.mdx:3]: I can't comment on things that weren't changed, but on line 89 the words "Role out" should be "Roll out"

Comment (danielleflinn) [reply libs/components/src/stories/Introduction.stories.mdx:3]: fixed, nice catch!

File: libs/components/src/stories/buttondocs.stories.mdx (hunk 5 above)
Comment (Hinton) [reply libs/components/src/stories/buttondocs.stories.mdx]: It's our naming convention for the repository.

**Review by Hinton — APPROVED** [review]

**Review by Hinton — APPROVED** [review]

**Pull request merged**